### ISSUE TITLE
Migration to morphia 2.0.x break data retrieval because of timezone assumptions

### ISSUE THREAD

**Issue by ppartarr** [MorphiaOrg/morphia#1504]
**Describe the bug**
When migrating from Morphia 1.6.x to 2.0.x the data from in Mongo was saved with a UTC timezone and adjusted to local machine date as described in #1218. In Morphia 2.0.x (even with legacy options activated as shown below) the dates are loaded as UTC which causes a mismatch

```
	MapperOptions mapperOptions = MapperOptions.legacy().build();
	Morphia.createDatastore(mongoClient, dbName, mapperOptions);
```

**To Reproduce**
1. Save data to Mongo using Morphia 1.6.x
2. Migrate to Morphia 2.0.x
3. Try retrieving data saved with Morphia 1.6.x
4. Mismatch in timezone assumption breaks retrieval based on date

**Expected behavior**
I'd expect legacy options in 2.0.x to keep the `MapperOptions` set to `SYSTEM_DEFAULT` rather than removing them altogether...

**Additional context**
Any example models, queries, and executable test cases you can supply will greatly help debugging your issue:

**Comment by evanchooly** [comment]
OK.  So that wasn't as smooth a transition as I'd thought it was going to be.  Those can definitely be resurrected.

**Comment by evanchooly** [comment]
It says 2.1.0 right now but i'll create a 2.0.2 later and back port it.

**Comment by ppartarr** [comment]
That would be great, thanks! :+1:

**Comment by evanchooly** [comment]
there should be a 2.0.2-SNAPSHOT available if you want to give that a try.  This is currently the only 2.0.2 issue but I'm ok with a single fix release if that's what's needed.  Otherwise, I'll take stock and see if there's anything else I might backport to 2.0.2 as well.

**Comment by ppartarr** [comment]
Thanks for the fix! Could you go ahead with the 2.0.2 release if possible?

**Comment by evanchooly** [comment]
You tried it then and it fixes your issue?  If so I'll try to get that released today.

**Comment by evanchooly** [comment]
Later than intended but I pushed 2.0.2 to central tonight.

**Comment by ppartarr** [comment]
Yes I did some quick testing and it looks like the issue is fixed with 2.0.2. Sorry for not getting back to you earlier. Thanks for the release! 😄
